chainermn/functions/batch_normalization.py

```diff
@@ -160,16 +160,28 @@ def forward(self, inputs):
             # Update running statistics:
             m = x.size // gamma.size
             adjust = m / max(m - 1., 1.)  # unbiased estimation
-            self.running_mean *= self.decay
-            temp_ar = xp.array(mean)
-            temp_ar *= (1 - self.decay)
-            self.running_mean += temp_ar
-            del temp_ar
-            self.running_var *= self.decay
-            temp_ar = xp.array(var)
-            temp_ar *= (1 - self.decay) * adjust
-            self.running_var += temp_ar
-            del temp_ar
+            if xp is numpy:
+                self.running_mean *= self.decay
+                temp_ar = xp.array(mean)
+                temp_ar *= (1 - self.decay)
+                self.running_mean += temp_ar
+                del temp_ar
+                self.running_var *= self.decay
+                temp_ar = xp.array(var)
+                temp_ar *= (1 - self.decay) * adjust
+                self.running_var += temp_ar
+                del temp_ar
+            else:
+                cuda.elementwise(
```

Comment:
syntax doesn't seem to be correct. maybe these lines should be changed to:
```
cuda.elementwise(
                    'T mean, T var, T decay, T adjust',
                    'T r_mean, T r_var',
                    '''
                    r_mean = r_mean * decay + mean * (1 - decay);
                    r_var = r_var * decay + var * (1 - decay) * adjust;
                    ''',
                    'update_mean_var'
```
?

Reply:
Thanks!

Reply:
I fixed.

Reply:
the error is not completely fixed. The comma after `T r_var` should be removed.
https://github.com/chainer/chainermn/pull/276/commits/8d37ea636aa9f5e37a029c39407e0efe6cdf3436#diff-3c18be054f7dc155b2c3b9bda576278bL177

Reply:
ohh, thanks.